### PR TITLE
Add cluster_upgrader builder

### PR DESCRIPTION
**Pull request by sedroche**

### CODE REVIEW

File: pkg/cluster_upgrader/cluster_upgrader.go [hunk 1]
```diff
@@ -67,29 +65,38 @@ const (
 // Interface describing the functions of a cluster upgrader.
 //go:generate mockgen -destination=mocks/cluster_upgrader.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/cluster_upgrader ClusterUpgrader
 type ClusterUpgrader interface {
-	UpgradeCluster(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) error
+	UpgradeCluster(upgradeConfig *upgradev1alpha1.UpgradeConfig, logger logr.Logger) error
+}
+
+//go:generate mockgen -destination=mocks/cluster_upgrader.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/cluster_upgrader ClusterUpgraderBuilder
+type ClusterUpgraderBuilder interface {
```
Comment (mrbarge) [inline pkg/cluster_upgrader/cluster_upgrader.go:72]: Has the mock been regenerated with this added interface? Not seeing it as part of the commit.

File: pkg/cluster_upgrader/cluster_upgrader.go [hunk 2]
```diff
@@ -715,29 +736,29 @@ func ValidateUpgradeConfig(c client.Client, m maintenance.Maintenance, upgradeCo
...
 	if versions[0] != current {
-		log.Info(fmt.Sprintf("validation failed, current version %s is greater than desired %s", current, upgradeConfig.Spec.Desired.Version))
+		logger.Info(fmt.Sprintf("validation failed, current version %s is greater than desired %s", current, upgradeConfig.Spec.Desired.Version))
 		metrics.UpdateMetricValidationFailed(upgradeConfig.Name)
 		return false, fmt.Errorf("desired version %s is greater than current version %s", upgradeConfig.Spec.Desired.Version, current)
 	}
 
 	// Find the available versions from cincinnati
-	uuid, err := uuid.Parse(string(clusterVersion.Spec.ClusterID))
+	uuID, err := uuid.Parse(string(clusterVersion.Spec.ClusterID))
```
Comment (mrbarge) [inline pkg/cluster_upgrader/cluster_upgrader.go:761]: _Extremely_ minor and pedantic nit - would prefer this stay as uuid - it's not really a convention to have the mixed case for the term.

File: pkg/cluster_upgrader/cluster_upgrader.go (hunk 1 above)
Comment (sedroche) [reply pkg/cluster_upgrader/cluster_upgrader.go:72]: I will regenerate once I start on fixing the tests. I wanted early feedback before I invest in the test :smile: 
Shall I commit them now so you can review?

File: pkg/cluster_upgrader/cluster_upgrader.go (hunk 2 above)
Comment (sedroche) [reply pkg/cluster_upgrader/cluster_upgrader.go:761]: This var clashes with the package name and my IDE suggested to fix it. maybe `clusterId`?

File: pkg/cluster_upgrader/cluster_upgrader.go (hunk 1 above)
Comment (mrbarge) [reply pkg/cluster_upgrader/cluster_upgrader.go:72]: Oh! No, that's fine, keep on going. :D  This all looks great to me.

File: pkg/cluster_upgrader/cluster_upgrader.go (hunk 2 above)
Comment (mrbarge) [reply pkg/cluster_upgrader/cluster_upgrader.go:761]: clusterId sounds good :+1:

**Pull request merged**